lib/webapp.js

```diff
@@ -928,6 +929,27 @@ WebServiceHandle.prototype = {
         requestOptions.headers = headers;
       }
 
+      while(this.service && this.service.type === 'import' && serviceModified) {
+        let reqDataservices;
+        serviceModified = false;
+        const reqPlugin = pluginsArray.filter(plugin => plugin.identifier === this.service.sourcePlugin);
+        reqPlugin.filter(plugin => {
+          reqDataservices = plugin.dataServices.filter(dataService => dataService.name === this.service.sourceName);
+        })
+        if(reqDataservices) {
+          const sourcePlugin = this.service.sourcePlugin;
+          const sourceName = this.service.sourceName;
+          this.service = {...reqDataservices[0], sourcePlugin: sourcePlugin, sourceName: sourceName, imported: true };
+          serviceModified = true;
+        }
+      }
+
+      if(this.service && this.service.imported) {
+        const reqPath = requestOptions.path.split('/').slice(6).join('/');
+        requestOptions.path = `/ZLUX/plugins/${this.service.sourcePlugin}/services/${this.service.sourceName}/${reqPath}`
```

Comment:
don't hardcode this logic as it could change and break later, use utility functions. Try using or expanding the content here https://github.com/zowe/zlux-server-framework/blob/343675ebff4b949b1a775dd9d45f2e5c0f7967c0/lib/url.js#L13-L30